packages/flame/lib/src/components/position_component.dart

```diff
@@ -135,20 +206,106 @@ abstract class PositionComponent extends BaseComponent {
   PositionComponent({
     Vector2? position,
     Vector2? size,
-    this.angle = 0.0,
+    double angle = 0.0,
     this.anchor = Anchor.topLeft,
     this.renderFlipX = false,
     this.renderFlipY = false,
     int? priority,
-  })  : _position = position ?? Vector2.zero(),
-        _size = size ?? Vector2.zero(),
-        super(priority: priority);
+  })  : _angle = angle,
+        _scaleX = 1.0,
+        _scaleY = 1.0,
+        _transformMatrix = Matrix4.identity(),
+        _recalculateTransform = true,
+        super(priority: priority) {
+    _position = _NotifyingVector2(this)..setFrom(position ?? Vector2.zero());
+    _size = _NotifyingVector2(this)..setFrom(size ?? Vector2.zero());
+  }
+
+  /// The total transformation matrix for the component. This matrix combines
+  /// translation, rotation and scale transforms into a single entity. The
+  /// matrix is cached and gets recalculated only as necessary.
+  Matrix4 get transformMatrix {
+    if (_recalculateTransform) {
+      final m = _transformMatrix.storage;
+      final cosA = math.cos(_angle);
+      final sinA = math.sin(_angle);
```

Comment:
Can't you use `_transformMatrix.rotate/translate/scale`?
I think readability here will give more than the small performance gain in not calling the `Matrix4` methods.

Reply:
I've added a comment to clarify what's going here. This way there are both the human-readable and CPU-optimized versions side by side. In addition, I added a test `'transform matrix'` to verify that these two forms are equivalent (except for the scale, the scale is currently not settable).

Hopefully, this would solve the readability concerns while retaining efficiency.

Reply:
That is a good middle ground. 

Were you able to join discord btw?
Then we could discuss how to tackle the merge conflicts that we will get between the two PRs.
https://discord.gg/qTexXyqR